versions/protocol-2-x.md

```diff
@@ -1465,6 +1466,123 @@ _Response_
+ */
+interface WorkspaceDocumentationParams {
+	/**
+	 * An optional non-empty query string, to be matched against the `name` of
+	 * `Documentation` in the workspace. For example, a user may query `HTTP Request`
+	 * and the server may respond with the `Documentation` named `Request`
+	 * which is itself a child of `Documentation` named `HTTP`.
+	 */
+	query?: string;
+
+	/**
+	 * An optional non-empty query string for the `Documentation` with the specified ID.
+	 */
+	queryID?: number | string;
+}
+```
+
+_Response_
+* result: `Documentation?` defined as follows:
+```typescript
+/**
+ * Represents documentation for a programming construct (variable, function,
+ * etc.) or group of programming constructs in a workspace (library, package,
+ * crate, module, etc.)
+ *
+ * The language server defines the exact structure of the documentation in a
+ * way that makes sense for the specific language and concepts being described.
+ */
+interface Documentation {
+	/**
+	 * A unique identifier for this `Documentation`. It exists to allow one
+	 * piece of `Documentation` to link to another piece of `Documentation` via
+	 * `InteractiveString`.
+	 */
+	id: number | string;
+
+	/**
+	 * A single-line label to display for this documentation in e.g. the index
+	 * of a book, e.g. the name of a group of documentation, the name of a
+	 * library, the signature of a function or class, etc.
+	 */
+	label: InteractiveString;
+
+	/**
+	 * Whether or not this documentation and its children describe a major
+	 * chapter and clients should e.g. display it and its children on a
+	 * dedicated page.
+	 */
+	chapter: boolean;
+
+	/**
+	 * A detailed multi-line string that contains documentation
+	 * A single-line label to display for this documentation in an index or
+	 * overview section, e.g. the name of a package, the signature of a
+	 * function or class, etc.
+	 */
+	detail: InteractiveString;
+
+	/**
+	 * Documentation that should be logically nested below this `Documentation`
+	 * itself. For example, `Documentation` may describe a class and have
+	 * `children` describing each method.
+	 */
+	children: Documentation[];
+}
+```
+Where `InteractiveString` is defined as follows:
+```typescript
+/**
+ * A MarkedString where some ranges of the string can be interactive by
+ * performing hover/definition/references requests to get more information.
+ */
+interface InteractiveString {
+	/**
+	 * The MarkedString being described.
+	 */
+	value: MarkedString;
+
+	/**
+	 * Ranges in the `value` string that a user could perform a
+	 * workspace/documentation request on with a `queryID?` equal to the
+	 * associated ID value in this mapping, in order to link one piece of
+	 * `Documentation` to another.
+	 */
+	documentation: { [{range: Range]: number | string };
+
+	/**
+	 * Ranges in the `value` string that a user could perform a
+	 * textDocument/hover request on to get more information.
+	 */
+	hovers: { [{range: Range]: TextDocumentPositionParams };

```

Comment:
How will `TextDocumentPositionParams` map to LSIF? One idea would be to add an edge to `resultSet`. `TextDocumentPositionParams` is currently not used in LSIF, and it's not clear how it should work in cross-repo scenarios.